packages/edit-site/src/components/dataviews-actions/index.js

```diff
@@ -41,3 +41,31 @@ export const useEditPostAction = () => {
 		[ history ]
 	);
 };
+
+export const useQuickEditPostAction = () => {
+	const history = useHistory();
+	return useMemo(
+		() => ( {
+			id: 'quick-edit-post',
+			label: __( 'Quick Edit' ),
+			isPrimary: true,
+			icon: settings,
+			isEligible( post ) {
+				if ( post.status === 'trash' ) {
+					return false;
+				}
+				// It's eligible for all post types except theme patterns.
+				return post.type !== PATTERN_TYPES.theme;
+			},
+			callback( items ) {
+				const post = items[ 0 ];
+				history.push( {
+					postId: post.id,
```

Comment:
We need to preserve the rest location params here.

Reply:
I'm always cautious about this, which part do you want to preserve?

Reply:
At that point we usually have the `layout` param and we can have the `activeView`, so I wouldn't expect any issues to preserve all `params`. If we want to be more strict about it, and update in the future if we add more needed query args, we can add only those I guess for now.

We need for example if we are in `drafts` and `quick edit`, to stay there. Right now it goes to the default view.

Reply:
Makes sense but I think we should be explicit about what we want to preserve in the url. There might be things that don't make sense (canvas for instance...). This is in general what causes random navigation issues like we had in the past.

Reply:
I updated this to preserve "activeView"